### ISSUE TITLE
Increment Variable

### ISSUE THREAD

**Issue by ElijahLynn** [fish-shell/fish-shell#2580]
Is there a way in Fish to increment a variable for use in a while loop. I didn't see anything in the documentation.

**Comment by ElijahLynn** [comment]
I keep trying variations of `set foo (math $foo + 1)` but haven't nailed it yet. I am guessing we still need to use `set` in some way.

**Comment by faho** [comment]
```fish
set noodle 1
while true
  echo $noodle
  set noodle (math $noodle + 1)
end
```

works for me.

**Comment by ridiculousfish** [comment]
If you find yourself incrementing a variable, consider if `seq` could be used instead. For example, to loop over the valid indexes in an array:

    set val alpha beta gamma
    for i in (seq (count $val))
        echo $i $val[$i]
    end

another common approach to iterate an array is to pop stuff off of the beginning:

    set val alpha beta gamma
    while count $val > /dev/null
        echo $val[1]
        set -e val[1]
    end

those might be simpler than whatever you're trying to do.

**Comment by ridiculousfish** [comment]
Closing since I think we have an answer

**Comment by ElijahLynn** [comment]
Thanks for confirming @faho If I do `set var (math $var + 1)` on a prompt I get:
 > "(standard_in) 1: syntax error"

But in a script it works, interesting. 

My use case was just to fill a file with lines until it reached exactly 50KB to estimate how many lines/objects I can put into an Akamai purge request to their API (limit is 50KB). I originally was trying to pass in an argument of 1,000 for instance. But I think I had parentheses around the test argument like this and turns out I think that was what was failing `while test ($i -lt 100)`. This ended up working:

```
set i 0
while test $i -lt 100
   echo 'foo' $i
   set i (math $i + 1)
end
```

But ended I ended up using filesize anyways. Here was my code I ended up going with. 
```
while test (du -b /tmp/ak | awk '{print $1}') -lt 50001
  echo (du -b /tmp/ak | awk '{print $1}') bytes
  echo "https://community.akamai.com/thread/2706" >> /tmp/ak; echo (wc -l /tmp/ak | awk '{print $1}' ) lines
end
```

Thanks @ridiculousfish, those examples may be helpful someday!
